### ISSUE TITLE
Receiving malformed bytes after the first 1/5 of a fader

### ISSUE THREAD

**Issue by JustSomeRandomUsername** [someweisguy/esp_dmx#6]
I am using this libary with an esp32 and a custom board to recieve dmx from an analog light desk and have stumbeled over some strange behavior. 
In the first 1/5 of a dmx channel I am recieving data but if the fader is at ~1/5 the esp already detects around 0 and if I try to receive dmx values higher then this I get "Received malformed byte at slot 2" (In this case I am sending of channel 1). If I go higher with the fader between 5/10 and 7/10 the esp is receiving dmx again and above that again the same errors.
I have the same behavior on different dmx channels.

I fiddled with the dmx config (baudrate, break_num, idle_num...) but that didn't seem to change anything.
Its definitly possible I made errors in building my dmx -> uart board (I am new to board design) but the fact that I can recieve data makes it feels like something is wrong in the software.

I used the "DMXRead" example in ArduinoIDE.
Any soft of help or hint into the right direction would be appreciated.

**Comment by someweisguy** [comment]
Thanks for the feedback!

It sounds like you are using the DMXRead example with a few modifications; namely the switch-case code in the [Packet Status](https://github.com/someweisguy/esp_dmx#packet-status) section of the readme. This would mean that you are getting a `DMX_ERR_IMPROPER_SLOT` status code. This error code gets assigned in the DMX interrupt handler as the result of the UART hardware interrupt status. The line is [here](https://github.com/someweisguy/esp_dmx/blob/f60d1c894052ce8c6f2ed8b3cf96f27201056363/src/dmx/intr_handlers.h#L149), if you're interested. The hardware interrupt that is triggered is the `UART_FRM_ERR_INT` (I call it `UART_INTR_FRAM_ERR` in this library) interrupt. The documentation isn't very clear on what causes this interrupt to fire. In my experimentation, I've found that missing start or stop bits in the DMX data is what causes this to fire. In other words, you _should_ be getting this error code due to the ESP32 hardware signaling that it is not getting the proper number of start or stop bits. 

To sum up the error you are receiving, you are saying that when your fader is between 0% to approximately 20%, you receive DMX data. When you set your fader to approximately 20% to 50%, you can no longer receive data. But between 50% and 70%, you receive data again. And then anything above 70%, you don't receive data.

A few questions for you:

1. Before you get the malformed slot error, is the DMX data that the ESP32 receives accurate? In other words, when you set the fader to 0 what DMX value are you receiving on that address? Is the data also 0, or is it some other value?
2. When you set your fader to approximately 20%, what is the last DMX value that the ESP32 receives before it reports the error?
3. Which lighting console are you using to send data to the ESP32?

**Comment by JustSomeRandomUsername** [comment]
Thanks for the long anwser, yes I added the switch statements to the read example.

To your questions:
1. from around 0% - 20% the values go down from 0xFE to around 0x00, its a bit hard to tell the exact end point but it looks like it goes towards 0x00 and the "overflows" or something.
2. around 0x00
3. its an old zero88 jester 12/24, https://zero88.com/control/jester

I should have time tomorrow to dive deeper into this, thanks for the help

**Comment by someweisguy** [comment]
Oh, cool console! 

I think the polarity of your DMX receiver board may be reversed. With the fader at 0%, I would expect to see 0x00 on the ESP32, not 0xFE. But if your polarity is reversed, it is possible for the ESP32 UART hardware to interpret 0x00 as 0xFE.

I would check to see if pins 2 and 3 are swapped in your XLR cable. If not, try swapping them. I've used alligator clips to do this. You can just unscrew the XLR connector and clip onto the appropriate pins where they are soldered, and then clip into you board..

**Comment by JustSomeRandomUsername** [comment]
Jup, I soldered the 2 data pins the wrong way round 🤦.
Work great now, thanks so much for the great help
